web3/eth/async_eth.py

```diff
@@ -522,6 +522,20 @@ async def get_storage_at(
         mungers=[BaseEth.filter_munger],
     )
 
+    # eth_sign_typed_data
+    async def sign_typed_data(self, account, message, domain):
```

Comment:
This should follow the pattern laid out already in the file, and should look something like:
```
_sign_typed_data: Method[
    Callable[[Union[Address, ChecksumAddress, ENS], str], HexStr]
] = Method(
    RPC.eth_signTypedData,
    mungers=[default_root_munger],
)
 
async def sign_typed_data(self, account, data):
    return await self.sign_typed_data(account, data)
```